contracts/amm/AmmSwapsLens.sol

```diff
@@ -0,0 +1,391 @@
+        internal
+        view
+        returns (IAmmSwapsLens.IporSwap[] memory swaps)
+    {
+        MiltonStorageTypes.IporSwapId[] memory swapIdsWithDirection = new MiltonStorageTypes.IporSwapId[](
+            swapIds.length
+        );
+        for (uint256 i = 0; i != swapIds.length; i++) {
+            swapIdsWithDirection[i] = MiltonStorageTypes.IporSwapId({id: swapIds[i], direction: 1});
+        }
+        return _mapSwaps(assetConfiguration, swapIdsWithDirection);
+    }
+
+    function _mapSwaps(AssetConfiguration memory assetConfiguration, MiltonStorageTypes.IporSwapId[] memory swapIds)
+        internal
+        view
+        returns (IAmmSwapsLens.IporSwap[] memory swaps)
+    {
+        IMiltonStorage miltonStorage = IMiltonStorage(assetConfiguration.miltonStorage);
+        uint256 accruedIbtPrice = iporOracle.calculateAccruedIbtPrice(assetConfiguration.asset, block.timestamp);
+        IAmmSwapsLens.IporSwap[] memory mappedSwaps = new IAmmSwapsLens.IporSwap[](swapIds.length);
+        for (uint256 i = 0; i != swapIds.length; i++) {
+            MiltonStorageTypes.IporSwapId memory swapId = swapIds[i];
+            if (swapId.direction == 0) {
+                IporTypes.IporSwapMemory memory iporSwap = miltonStorage.getSwapPayFixed(swapId.id);
+                int256 swapValue = iporSwap.calculatePayoffPayFixed(block.timestamp, accruedIbtPrice);
+                mappedSwaps[i] = _mapSwap(assetConfiguration.asset, iporSwap, 0, swapValue);
+            } else {
+                IporTypes.IporSwapMemory memory iporSwap = miltonStorage.getSwapReceiveFixed(swapId.id);
+                int256 swapValue = iporSwap.calculatePayoffReceiveFixed(block.timestamp, accruedIbtPrice);
+                mappedSwaps[i] = _mapSwap(assetConfiguration.asset, iporSwap, 1, swapValue);
+            }
+        }
+        return mappedSwaps;
+    }
+
+    function _mapSwap(
+        address asset,
+        IporTypes.IporSwapMemory memory swap,
+        uint256 direction,
+        int256 swapValue
+    ) internal pure returns (IAmmSwapsLens.IporSwap memory) {
+        return
+            IAmmSwapsLens.IporSwap({
+                id: swap.id,
+                asset: asset,
+                buyer: swap.buyer,
+                collateral: swap.collateral,
+                notional: swap.notional,
+                leverage: IporMath.division(swap.notional * Constants.D18, swap.collateral),
+                direction: direction,
+                ibtQuantity: swap.ibtQuantity,
+                fixedInterestRate: swap.fixedInterestRate,
+                payoff: swapValue,
+                openTimestamp: swap.openTimestamp,
+                endTimestamp: swap.endTimestamp,
+                liquidationDepositAmount: swap.liquidationDepositAmount,
+                state: swap.state
+            });
+    }
+
+    function owner() public view virtual returns (address) {
```

Comment:
This should be removed, i think ?

Reply:
here we should use library which read from slot where owner address is stored in router

Reply:
removed as agreed